### PR TITLE
Fix vevent location backcompat

### LINKED ISSUE #184: Fix vevent location backcompat
From https://github.com/microformats/tests/blob/master/tests/microformats-v1/hcalendar/combining.html

```html
<div class="vevent">
  <a class="summary url" href="http://indiewebcamp.com/2012">
    IndieWebCamp 2012
  </a>
  from <time class="dtstart">2012-06-30</time> 
  to <time class="dtend">2012-07-01</time> at 
  <span class="location vcard">
    <a class="fn org url" href="http://geoloqi.com/">Geoloqi</a>, 
    <span class="adr">
        <span class="street-address">920 SW 3rd Ave. Suite 400</span>, 
        <span class="locality">Portland</span>, 
        <abbr class="region" title="Oregon">OR</abbr>
    </span>
  </span>
</div>
```

Currently parses the vcard as a child instead of location property:

```json
"items": [
    {
        "type": [
            "h-event"
        ],
        "properties": {
            "name": [
                "IndieWebCamp 2012"
            ],
            "url": [
                "http://indiewebcamp.com/2012"
            ],
            "start": [
                "2012-06-30"
            ],
            "end": [
                "2012-07-01"
            ]
        },
        "children": [
            {
                "type": [
                    "h-card"
                ],
                "properties": {
                    "name": [
                        "Geoloqi"
                    ],
                    "org": [
                        "Geoloqi"
                    ],
                    "url": [
                        "http://geoloqi.com/"
                    ],
                    "adr": [
                        {
                            "type": [
                                "h-adr"
                            ],
                            "properties": {
                                "street-address": [
                                    "920 SW 3rd Ave. Suite 400"
                                ],
                                "locality": [
                                    "Portland"
                                ],
                                "region": [
                                    "Oregon"
                                ]
                            },
                            "value": "920 SW 3rd Ave. Suite 400, \n        Portland, \n        Oregon"
                        }
                    ]
                }
            }
        ]
    }
]
```

I think [this line](https://github.com/indieweb/php-mf2/blob/master/Mf2/Parser.php#L1572) needs to be changed to `p-location h-card`.

### PR DESCRIPTION
**Pull request by gRegorLove**
Includes whitespace fixes in ClassicMicroformatsTest.php [View updates sans whitespace](https://github.com/microformats/php-mf2/pull/201/files?w=1).

Fixes #184

### CODE REVIEW

**Comment by Zegnat** [comment]
According to the original issue #184 this tries to address two tests:

1. ✅ [hcalendar/combining](https://github.com/microformats/tests/blob/master/tests/microformats-v1/hcalendar/combining.html): this one now passes. No more extra `children` property. `location` is correctly added with a nested `h-card`.
2. ❌ [hcalendar/attendees](https://github.com/microformats/tests/blob/master/tests/microformats-v1/hcalendar/attendees.html): this one does not pass yet. This PR now adds a `location` property, the test expect it to be the string `San Francisco` but the parser now makes it an `h-card`. If there is no `vcard` or `adr` class, none should be implied.

File: Mf2/Parser.php [hunk 1]
```diff
@@ -1578,7 +1578,7 @@ public function backcompat(DOMElement $el, $context = '', $isParentMf2 = false)
 					if ( $location->length ) {
 						foreach ( $location as $tempEl ) {
 							if ( !$this->hasRootMf2($tempEl) ) {
-								$this->addMfClasses($tempEl, 'h-card');
+								$this->addMfClasses($tempEl, 'p-location h-card');
 								$this->backcompat($tempEl, 'vcard');
```
Comment (Zegnat) [inline Mf2/Parser.php:1582]: Needs some type of `if` in here somewhere.

If the location element does not have and `adr` or `vcard` class it needs to be handled as a plain `p-location` string. I _think_ `adr` and `vcard` are both parsed into an h-card, if I am mistaken those two cases also need separate handling.

**Comment by gRegorLove** [comment]
The Travis failure now is on PHP 5.5 with masterminds/html5. Once #220 is resolved I can rebase on master and that should be good to go.

**Comment by gRegorLove** [comment]
The [.editorconfig](https://github.com/microformats/php-mf2/blob/master/.editorconfig) introduced a little bit ago specifies tabs so I think my editor did that automatically.

Adding `?w=1` to the URL removes the whitespace changes for reviewing: https://github.com/microformats/php-mf2/pull/201/files?w=1

**Review by Zegnat — APPROVED** [review]
I have rerun the tests with the new testing in place on master.

```
composer run phpunit -- --group microformats/tests/mf1
```

With this PR merged ([like before](https://github.com/microformats/php-mf2/pull/201#issuecomment-622939429)), we are now able to pass the hcalendar/combining test.

But also like before, this still does not fix hcalendar/attendees:

```
Mf2\Parser\Test\MicroformatsTestSuiteTest::testMf1FromTestSuite with data set "hcalendar/attendees" ('<meta charset="utf-8">\n<div c.../div>\n', '{\n    "items": [{\n        "ty...: {}\n}', 'hcalendar/attendees')
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
     'items' => Array (
         0 => Array (
             'properties' => Array (
-                'attendee' => Array (...)
                 'location' => Array (...)
                 'name' => Array (...)
                 'start' => Array (...)
             )
             'type' => Array (...)
+            'children' => Array (...)
         )
     )
     'rel-urls' => Array ()
     'rels' => Array ()
 )
```

This PR does not introduce any new test breaks, so it looks good to me for merge. But we need to decide whether we want the hcalendar/attendees test to pass or not.

Would be nice to get this branch rebased on master so it is easier to run tests.

**Pull request closed without merging**